sw/device/lib/runtime/otbn.c

```diff
@@ -195,3 +196,16 @@ otbn_result_t otbn_zero_data_memory(otbn_t *ctx) {
   }
   return retval;
 }
+
+void otbn_dump_dmem(const otbn_t *ctx, uint32_t max_addr) {
+  if (max_addr == 0) {
+    max_addr = dif_otbn_get_dmem_size_bytes(&ctx->dif);
+  }
+
+  uint32_t data[8];
```

Comment:
Nit: Move this closer to where it is first used? (LLVM will LICM it out anyways.)

Reply:
Why would that be (substantially) better?

I find it close enough to the site where it's used (we're talking at most 2 lines!), and having it outside the loop additionally makes it clear that I want at most one allocation.

Reply:
I agree with @mcy on this one, `data` should be inside the loop so that it is scoped as locally as possible. That makes it clearer that the contents of `data` are not transferred from one iteration to the next or outside of the loop. I think the only performance reason to have it outside the loop would be if it needed initialization.